### PR TITLE
add search on content list

### PR DESCRIPTION
**Pull request by veolys-nanne**
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/722
https://github.com/open-orchestra/open-orchestra-model-interface/pull/280
https://github.com/open-orchestra/open-orchestra-mongo-libs/pull/66
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/2068
https://github.com/open-orchestra/open-orchestra-libs/pull/74
https://github.com/open-orchestra/open-orchestra/pull/1131

### CODE REVIEW

File: Pagination/MongoTrait/FilterTypeStrategy/Strategies/DateFilterStrategy.php [hunk 1]
```diff
@@ -29,7 +42,11 @@ public function support($type)
      */
     public function generateFilter($name, $value, $documentName)
     {
-        $value = str_replace("/", "-", trim($value));
+        if ($this->contextManager->getDefaultLocale() == 'en') {
```
Comment (alavieille) [inline Pagination/MongoTrait/FilterTypeStrategy/Strategies/DateFilterStrategy.php]: and the other local which use the same syntax ?

**Review by alavieille — CHANGES_REQUESTED** [review]

File: Pagination/MongoTrait/FilterTrait.php [hunk 2]
```diff
@@ -26,228 +25,38 @@ public function setFilterTypeManager(FilterTypeManager $filterTypeManager)
...
-     * @param Stage               $qa
-     * @param FinderConfiguration $configuration
+     * @param PaginateFinderConfiguration $configuration
+     * @param Stage                       $qa
+     * @param array                       $searchTypes
      *
      * @return Stage
      */
-    protected function generatePreFilter(Stage $qa, FinderConfiguration $configuration)
+    protected function generateFieldFilter(PaginateFinderConfiguration $configuration, Stage $qa, array $searchTypes)
```
Comment (ngilain) [inline Pagination/MongoTrait/FilterTrait.php]: @itkg-nanne : should introduce plural form in the function name as many filters are generated

**Comment by veolys-nanne** [comment]
update done @alavieille , @ngilain

**Comment by ngilain** [comment]
:+1:

File: Pagination/MongoTrait/FilterTypeStrategy/Strategies/StringFilterStrategy.php [hunk 3]
```diff
@@ -17,17 +17,18 @@ class StringFilterStrategy implements FilterTypeInterface
      */
     public function support($type)
     {
-        return $type === 'string';
+        return $type === 'text';
```
Comment (alavieille) [inline Pagination/MongoTrait/FilterTypeStrategy/Strategies/StringFilterStrategy.php]: can you use const for this ?

**Review by alavieille — CHANGES_REQUESTED** [review]

**Comment by veolys-nanne** [comment]
update done @alavieille

**Review by alavieille — APPROVED** [review]

**Pull request merged**